packages/suite/src/utils/wallet/coinjoinUtils.ts

```diff
@@ -61,6 +61,45 @@ export const breakdownCoinjoinBalance = ({
     return balanceBreakdown;
 };
 
+/**
+ * Calculates account anonymity progress – how much UTXOs are anonymized relatively to the target anonymity
+ */
+export const calculateAnonymityProgress = ({
+    targetAnonymity,
+    anonymitySet,
+    utxos,
+}: {
+    targetAnonymity: number | undefined;
+    anonymitySet: AnonymitySet | undefined;
+    utxos: Account['utxo'];
+}): number => {
+    if (!anonymitySet || targetAnonymity === undefined || !utxos) {
+        return 0;
+    }
+
+    let weightedCurrentAnonymity = new BigNumber(0);
+    let weightedTargetAnonymity = new BigNumber(0);
+
+    utxos?.forEach(({ address, amount }) => {
+        const bigAmount = new BigNumber(amount);
+
+        const currentAnonymity = anonymitySet[address] || 0;
```

Comment:
0 would go to -1 on L89
```suggestion
        const currentAnonymity = anonymitySet[address] || 1;
```

Reply:
f56eaa083acc69a8652848b738da50547c3beaa6